jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/QueryServiceImpl.java

```diff
@@ -232,20 +223,24 @@ public void unregisterQuery(final String uniqueQueryName) throws QueryNotFoundEx
             // call builder again in case more parameters are available
             filter = paramBuilder.build();
         }
-        
-        if (queryContext.getOrderBy() != null) {
-            String[] oderByItems = queryContext.getOrderBy().split(",");
-            
-            for (String orderBy : oderByItems) {
-                logger.debug("Applying order by {} and ascending {}", orderBy, queryContext.isAscending());
-                builder.sort(orderBy.trim(), queryContext.isAscending()?"asc":"desc");
+
+        // process the ORDER BY clause into order by and sort order pairs
```

Comment:
if we will use the QueryContext as method signature then we could just add a check here to handle orderByClause in the actual impl is AdvancedQueryContext